<?xml version="1.0"?>
<launch>

<!-- what the launch statement should look like -->
<!-- /home/tractor/ros1_lawn_tractor_ws/src/pure_pursuit/launch/pure_pursuit_cub_cadet_oct23.launch -->
<!-- roslaunch pure_pursuit pure_pursuit_cub_cadet_oct23.launch origin_lat:=40.712776 origin_lon:=-74.005974 ref_lat:=40.345308 ref_lon:=-80.128855 -->

  <node pkg="pure_pursuit" type="pure_pursuit" name="controller"  clear_params="true">
    <param name="robot_frame_id" value="base_link"/>
    <!-- <param name="robot_frame_id" value="base_footprint"/>  -->
    <!-- change to base_link instead of base_footprint -->
    <param name="global_frame_id" value="odom"/>
    <param name="lookahead_frame_id" value="lookahead"/>
    <rosparam file="$(find pure_pursuit)/cfg/pure_pursuit_cub_cadet.yaml" command="load"/>    
    <remap from="path_segment" to="/drive_path"/>
    <remap from="odometry" to="odom"/>
    <remap from="cmd_vel" to="cmd_vel"/>
    <remap from="cmd_acker" to="/acker_in"/>
  </node>  

      <!-- Define required launch file input arguments for path_update_origin_lat_lon  -->
    <arg name="origin_lat"/>
    <arg name="origin_lon"/>
    <arg name="ref_lat"/>
    <arg name="ref_lon"/>

    <!-- Set ROS parameters from arguments that will be read as input in the script path_update_origin_lat_lon  -->
    <param name="GPS_origin_lat" value="$(arg origin_lat)"/>
    <param name="GPS_origin_lon" value="$(arg origin_lon)"/>
    <param name="GPS_ref_lat" value="$(arg ref_lat)"/>
    <param name="GPS_ref_lon" value="$(arg ref_lon)"/>

    <!-- Start the Python Script Node -->
<!--     <node name="path_update_origin_lat_lon" pkg="ackermann_vehicle" type="path_update_origin_lat_lon.py" output="screen">
    </node> -->

  </launch>